e2e/tests/helpers/mod.rs

```diff
@@ -223,8 +230,22 @@ impl Api {
 
 impl Drop for Api {
     fn drop(&mut self) {
-        if let Some(server) = self.server.as_mut() {
-            server.process.kill().unwrap();
+        if let Some(container) = &self.container {
+            Command::new("docker")
+                .args(["stop", container])
+                .output()
+                .expect("failed to stop api container");
```

Comment:
Oh, if `stop` fails then so will `rm`. And if `rm` fails, then so will `rmi`. So maybe we want to know where the failures started (panic) for manual cleanup?

Reply:
Ah, I misunderstood! `output()` will be `Err` if the process fails to start, not if it fails to run. I thought this would panic early if `docker stop` fails to run. I gotchu 👍